src/components/ui/time-picker.svelte

```diff
@@ -0,0 +1,121 @@
+<script>
+  import { createEventDispatcher } from 'svelte';
+  import Button from './button.svelte';
+  import TextField from './text-field.svelte';
+
+  export let value = { hours: null, minutes: null };
+  $: {
+    if (value == null) {
+      value = { hours: null, minutes: null };
+    }
+  }
+  const minutesInHour = 60;
+  const hoursInDay = 24;
+
+  const dispatch = createEventDispatcher();
+
+  function incHours() {
+    if (value.hours == null) {
+      value.hours = 0;
+    } else {
+      value.hours++;
+      if (value.hours > hoursInDay - 1) {
+        value.hours = 0;
+      }
+    }
+
+    if (value.hours != null && value.minutes != null) {
+      dispatch('change', value);
+    }
+  }
+
+  function decHours() {
+    if (value.hours == null) {
+      value.hours = 0;
+    } else {
+      value.hours--;
+      if (value.hours < 0) {
+        value.hours = hoursInDay - 1;
+      }
+    }
+
+    if (value.hours != null && value.minutes != null) {
+      dispatch('change', value);
+    }
+  }
```

Comment:
A lot of repetition. Same exact code in `incHours`, except for the `-` instead of `+`. Additionally, `null` is treated as `0` in math operations. I suggest combining into one function with a parameter, and perhaps 2 auxiliary functions for the template section:
```js
function changeHours(offset) {
  value.hours = (value.hours + offset) % hoursInDay;
  if (value.minutes != null)
    dispatch('change', value);
}

const incHours = () => changeHours(1);
const decHours = () => changeHours(-1);
```